src/main/java/slimeknights/tconstruct/library/tools/item/ranged/ModifiableLauncherItem.java

```diff
@@ -386,8 +402,99 @@ public boolean onBlockStartBreak(ItemStack stack, BlockPos pos, Player player) {
+  public ProjectileData createProjectile(Level level, LivingEntity user, ItemStack ammo) {
+    ArrowItem arrowItem = ammo.getItem() instanceof ArrowItem a ? a : (ArrowItem) Items.ARROW;
+    AbstractArrow arrow = arrowItem.createArrow(level, ammo, user);
+    return new ProjectileData(arrow, 1f, 1);
+  }
+
+  /** Modify projectile launch angle for specific angle offset. Bow use vertical spread and crossbow use horizontal spread. */
+  public abstract Vector3f modifyShootAngle(LivingEntity user, Vec3 target, float angle);
+
+  /**
+   * Launch projectile, play sound, and optionally consume durability. Does not handle ammunition consumption.
+   *
+   * @param holder           user
+   * @param tool             projectile weapon
+   * @param hand             hand. For weapon damaging only.
+   * @param ammo             ammunition. Will not be modified inside this method.
+   * @param direction        direction to shoot at
+   * @param charge           charging percentage
+   * @param speedFactor      regular arrow speed at full charge. 3 for player with bow. 1.6 for skeletons. 3.15 for crossbows.
+   * @param inaccuracyFactor shoot inaccuracy factor. 1 for player, and hostile mobs use a complex formula involving difficulty.
+   * @param infinite         Whether to make the arrow marked as no-pickup. True for infinite arrows or hostile mob arrows.
+   * @param damageWeapon     Whether to damage weapon. False for hostile mobs.
+   */
+  public void shootProjectiles(LivingEntity holder, IToolStackView tool, InteractionHand hand, ItemStack ammo, Vec3 direction, float charge,
+                               float speedFactor, float inaccuracyFactor, boolean infinite, boolean damageWeapon) {
+    float velocity = ConditionalStatModifierHook.getModifiedStat(tool, holder, ToolStats.VELOCITY);
+    float inaccuracy = ModifierUtil.getInaccuracy(tool, holder);
+    float startAngle = getAngleStart(ammo.getCount());
+    int primaryIndex = ammo.getCount() / 2;
+    int damage = 0;
+    for (int arrowIndex = 0; arrowIndex < ammo.getCount(); arrowIndex++) {
+      // setup projectile
+      ProjectileData data = createProjectile(holder.level(), holder, ammo);
+      Projectile projectile = data.projectile();
+      damage += data.damage();
+      assignArrowProperties(projectile, holder, tool, charge >= 1, infinite, arrowIndex == primaryIndex);
+      // setup projectile
+      float angle = startAngle + (10 * arrowIndex);
+      Vector3f targetVector = modifyShootAngle(holder, direction, angle);
+      projectile.shoot(targetVector.x(), targetVector.y(), targetVector.z(),
+        charge * velocity * data.speed() * speedFactor,
+        inaccuracy * inaccuracyFactor);
+      // finally, fire the projectile
+      holder.level().addFreshEntity(projectile);
+      playShotSound(holder, charge, angle, holder.getRandom());
+    }
+    if (damageWeapon) {
+      ToolDamageUtil.damageAnimated(tool, damage, holder, hand);
+    }
```

Comment:
Not sure on this point, but I do wonder whether it wouldn't be simpler to just `return damage` and let the caller handle it/ignore it, lets us drop 2 parameters from the method as hand is only needed for this method. Also gives me freedom to have say, a helmet shoot arrows with the helmet keybind.

Reply:
Maybe I can do, `shootProjectiles` and `shootProjectilesAndDamageWeapon`?

Reply:
Its not hard to write:
```java
int damage = shootProjectiles(...);
if (!creative) {
  ToolDamageUtil.damageAnimated(...);
}
```
so I think just `shootProjectiles` is sufficient

Reply:
It fool-proves addons who would have called `stack.hurtAndBreak`

Reply:
Hurt and break is fine for them to call, its just less efficient as it requires reparsing NBT.

And to be blunt, if an addon calling *Tinkers' Construct code* doesn't realize they should call *Tinkers' Construct methods* while using an internal API, there is not much I can do for them.

Point of the original comment was to minimize complexity of the API by separating responsibilities, not make it more complex by adding similar but slightly different methods.

Reply:
ok. I will remove the redundant method